src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -247,4 +251,56 @@ tsi_result SslProtectorUnprotect(const unsigned char* protected_frames_bytes,
   return result;
 }
 
+int VerifyCrlSignature(X509_CRL* crl, X509* issuer) {
+  if (issuer == nullptr || crl == nullptr) {
+    return -1;
+  }
+  EVP_PKEY* ikey = X509_get_pubkey(issuer);
+  if (ikey == nullptr) {
+    // Can't verify signature because we couldn't get the pubkey, fail the
+    // check.
+    EVP_PKEY_free(ikey);
+    return 1;
+  }
+  int ret = X509_CRL_verify(crl, ikey);
+  EVP_PKEY_free(ikey);
+  return ret;
+}
+
+int VerifyCrlCertIssuerNamesMatch(X509_CRL* crl, X509* cert) {
+  if (cert == nullptr || crl == nullptr) {
+    return 1;
+  }
+  X509_NAME* cert_name = X509_get_issuer_name(cert);
+  if (cert == nullptr) {
+    return 1;
+  }
+  X509_NAME* crl_issuer_name = X509_CRL_get_issuer(crl);
+  if (crl_issuer_name == nullptr) {
+    return 1;
+  }
+  return X509_NAME_cmp(cert_name, crl_issuer_name);
+}
+
+bool VerifyCrlSignBit(X509* issuer) {
```

Comment:
I guess the rationale to use `issuer` here is that you are expected to pass the cert for the issuer of some crl to this API, whereas for `VerifyCrlCertIssuerNamesMatch` you'd be expected to pass in a cert and crl that were both issued by the same CA, WDYT

I think it could help someone reading the code differentiate which specific cert in the chain they are supposed to pass?

Reply:
I think what you describe is one use case, but someone in the future could be interested in checking if their CA cert has this bit set, in which case they should not be looking at the issuer of the CA cert.

To that end, WDYT of changing the signature to:
```
bool HasCrlSignBit(X509* cert);
```

Reply:
That sounds good